File: 13-15/JAVA/src/animals/domestic/DomesticAnimal.java
// DomesticAnimal.java
package animals.domestic;

import animals.Animal;

public class DomesticAnimal extends Animal {
    public DomesticAnimal(String name, String birthDate) {
        super(name, birthDate);
    }
}

// Dog.java
package animals.domestic;

public class Dog extends DomesticAnimal {
    public Dog(String name, String birthDate) {
        super(name, birthDate);
    }
}

// Cat.java
package animals.domestic;

public class Cat extends DomesticAnimal {
    public Cat(String name, String birthDate) {
        super(name, birthDate);
    }
}

// Hamster.java
package animals.domestic;

public class Hamster extends DomesticAnimal {
    public Hamster(String name, String birthDate) {
        super(name, birthDate);
    }
}

File: 13-15/C++/src/DomesticAnimal.cpp
#include "DomesticAnimal.h"

DomesticAnimal::DomesticAnimal(const std::string &name, const std::string &birthDate)
    : Animal(name, birthDate) {}

#include "Dog.h"
Dog::Dog(const std::string &name, const std::string &birthDate)
    : DomesticAnimal(name, birthDate) {}

#include "Cat.h"
Cat::Cat(const std::string &name, const std::string &birthDate)
    : DomesticAnimal(name, birthDate) {}

#include "Hamster.h"

Hamster::Hamster(const std::string &name, const std::string &birthDate)
    : DomesticAnimal(name, birthDate) {}
